library/src/main/java/org/apache/apex/malhar/lib/dedup/TimeBasedDedupOperator.java

```diff
@@ -186,7 +186,7 @@ public void deactivate()
   @Override
   protected void putManagedState(Object tuple)
   {
-    ((ManagedTimeUnifiedStateImpl)managedState).put(getTime(tuple), getKey(tuple), new Slice(new byte[0]));
+    ((ManagedTimeUnifiedStateImpl)managedState).put(getTime(tuple), getKey(tuple), new Slice(null, 0, 0));
```

Comment:
The change is not required for the upgrade. Open another JIRA and PR to review Slice usage (including how empty Slices are constructed).